### DISCUSSION TITLE
How do you validate initial default values?

### DISCUSSION THREAD

**Discussion by thomaschaaf** [react-hook-form/react-hook-form#2348] [Q&A]
Is the correct way to validate the defaultValues at the start using a useEffect using `trigger` like this? Or is there another way to trigger the validation of the defaultValues?

https://codesandbox.io/s/react-hook-form-basic-validation-mtbz6?file=/src/index.js:385-476

**Comment by bluebill1049** [comment]
that's correct, or you can set mode to `onChange` and reading isValid formState

**Reply by jenyckee** [reply]
This doesn't provide you with the errors though

**Comment by bezenson** [comment]
For me `formState.isValid` works without anything else. I hope it's expected :)
